InteractiveHtmlBom/ecad/generic.py

```diff
@@ -0,0 +1,171 @@
+
+    def getBom(self):
+        self.bom = {'B': [], 'F': [], 'both': [], 'skipped': []}
+
+        for row in self.conn.execute(
+                '''SELECT footprint, value, side, count(*) AS qty FROM bom
+                   GROUP BY side, footprint, value'''):
+            bom_group = [
+                row['qty'],
+                str(row['value']),
+                str(row['footprint']),
+                [[str(r['refdes']), r['id']] for r in self.conn.execute(
+                    '''SELECT bom.refdes, xy.rowid - 1 AS id FROM bom, xy
+                     WHERE bom.refdes = xy.refdes AND footprint=? AND value=?''',
+                    (row['footprint'], row['value']))], []
+            ]
+
+            self.bom['both'].append(bom_group)
+            self.bom[row['side']].append(bom_group)
+
+    def getComponents(self):
+        self.components = []
+
+        for row in self.conn.execute(
+                'SELECT refdes, footprint, value, side FROM bom'):
+            self.components.append(
+                Component(row['refdes'], str(row['value']), str(
+                    row['footprint']), str(row['side'])))
+
+    def getModules(self):
+        self.modules = []
+
+        for row in self.conn.execute('SELECT refdes, x, y, side FROM xy'):
+            self.modules.append({
+                'center': [row['x'], row['y']],
+                'bbox': {
+                    'angle': 0,
+                    'pos': [row['x'], row['y']],
+                    'relpos': [self.BBOX_SIZE * -0.5, self.BBOX_SIZE * -0.5],
+                    'size': [self.BBOX_SIZE, self.BBOX_SIZE],
+                },
+                'pads': [],
+                'drawings': [],
+                'layer': row['side'],
+                'ref': str(row['refdes']),
+            })
+
+    def getBackground(self):
+        self.silkscreen = dict(F=[], B=[])
+
+        basename = os.path.splitext(self.file_name)[0]
+
+        self.silkscreen['F'].append({
+            "start": [0, 0],
+            "type": "url",
+            "url": basename + "-front.png"
+        })
+
+        self.silkscreen['B'].append({
+            "start": [0, 0],
+            "type": "url",
+            "url": basename + "-back.png"
+        })
+
+    def parse(self):
+
+        self.getBackground()
+        self.parseXy()
+        self.getBom()
+        self.getModules()
+
+        pcbdata = {
+            "edges_bbox": {
+                "minx": 0,
+                "miny": 0,
+                "maxx": self.WIDTH / self.MPP,
+                "maxy": self.HEIGHT / self.MPP
+            },
+            "edges": [],
+            "silkscreen":
+            self.silkscreen,
+            "fabrication": {
+                'F': [],
+                'B': [],
+            },
+            "modules":
+            self.modules,
+            "metadata": {
+                "title": "unamed",
+                "company": "unamed",
+                "revision": "v0.0",
+                "date": ""
```

Comment:
Suggestion: use file name and date for title and date, leave company and revision empty.

Reply:
Fixed.